### ISSUE TITLE
Should filter values be connected by AND or OR?

### ISSUE THREAD

**Issue by christophfriedrich** [Open-EO/openeo-hub#35]
E.g. when selecting multiple output formats: Should the UI show the backends that support *all* or *at least one* of them?

**Use Case for AND:** User needs to feed the result into multiple systems that expect different formats -- e.g. the result is needed as a GeoJSON, but for a quick overview there should also be a rendered JPEG image.

**Use Case for OR:** User has a system that understands multiple formats and just inputs them all to check whether there's one that is supported (but doesn't really care which one it is) -- e.g. the result is needed as an image, but whether that's JPEG or PNG doesn't really matter.

So far I implemented it with AND, but while implementing "filter by API version" (see upcoming commit), I realised that OR might be just as reasonable.

:thinking:

**Comment by m-mohr** [comment]
Could you have a checkbox with label 'all' (=and) per criteria? I guess both use cases are valid and should be catered for... 🤔

**Comment by christophfriedrich** [comment]
Such a checkbox is certainly possible, but rather untypical. It would make the UI a lot more technical. I was hoping there is another option? :thinking:

If we go that way, I would suggest two radioboxes with "and" and "or" as labels?

Or a toggle like this?
![grafik](https://user-images.githubusercontent.com/17781682/64597984-30a36000-d3b7-11e9-8c14-75485bd46300.png)

**Comment by m-mohr** [comment]
I would hope there's one, but I couldn't find any yet. 
I also thought about a toggle like that, but feel like and/or is even more technical than any/all. I don' have any proof though and would by happy with both using the toggle. The reason for proposing the checkbox was simply that it would take approx. half of the space.

**Comment by m-mohr** [comment]
@christophfriedrich Could you document here what we have decided on for each field? I think we discussed and decided on AND / OR for each field depending on the most likely use-case, right?
Processes were likely AND, Collections OR, ...
Maybe we only need this "switch" behavior only for a few selected fields or no fields at all?

**Comment by christophfriedrich** [comment]
Exactly, we went through all the fields and guessed what would be the most common use case. This is what we decided:

- API versions: OR
- endpoints: AND
- collections: OR
- processes: AND
- output formats: OR
- service types: OR

This is documented with comments in the corresponding source code:
https://github.com/Open-EO/openeo-hub/blob/7c852982aef96e62cdf78e8ca67e5f3b72d968ca/src/components/DiscoverSection.vue#L225-L253

IMO "potential switchable fields" are: collections, output formats, service types (because you may need more than 1 dataset for your calculations and may want to output in more than 1 format/to more than 1 service). Processes maaay be another candidate, but not _really_.

**Comment by christophfriedrich** [comment]
This issue can be found again one level deeper: **When filtering for functionalities/endpoints, how should incomplete functionalities be treated?** I.e. should the required endpoints _within one functionality_ be combined with AND or OR (i.e. ALL or ANY OF)? I.e. should functionalities for which the SupportedFeatures component (from openeo-vue-component) displays a :warning: be filtered out or left in?

One could argue that if you select the functionality, you just want it to work -- not only half of it. On the other hand, half the functionality may bring you further than if it doesn't work at all, so you may want to see that backend in the result anyway.

In this decision it should also be considered how strict the featurelist.js file defines a "full" implementation of a functionality. E.g. when GET and POST are sufficient for "full" (as is the case at the moment of writing this) AND may be okay, but if PATCH is also required then OR could be more appropriate.

**One could implement a switch for this too.**
